crux-mir/README.md

```diff
@@ -147,5 +234,111 @@ unsupported functions still remain.  Test cases that call into unsupported code
 will produce assertion failures with messages like `don't know how to call
 core::intrinsics::transmute`.
 
+
+### References & function pointers
+
 Currently, `crux-mir` also has trouble supporting references and function
 pointers in constant expressions and static initializers.
+
+
+### Symbolic size arrays
+
+`crux-mir` does support vectors (and data structures) with a symbolic capacity if an upper bound on the size is provided. Truly unbounded sizes are not supported. For example, a `Vec` with *symbolic* length but *fixed* capacity can be used as follows:
+
+```Rust
+let mut v = Vec::with_capacity(8);
+for _ in 0 .. usize::symbolic_where("n", |n| n <= 8) {
+    v.push(i32::symbolic("x"));
+}
+```
+
+
+### Don't know how to call...
+
+If you get a translation error and ` Don't know how to call...` message (an example is below), that usually means that your code contains a compiler intrinsic that `crucible-mir` doesn't know how to simulate. Often times, this can be fixed by adding a custom override for the intrinsic. In such case, we encourage you to look through the [existing issues](https://github.com/GaloisInc/crucible/issues) if such missing case was reported already, and if not, open a new issue.
+
+The error will look similar to this:
+
+```
+[Crux]   Translation error in subtle/81045fbd::black_box[0]::_inst2efc261c2cb07b6c[0]: callExp: Don't know how to call core/c7248340::intrinsics[0]::volatile_load[0]::_inst2efc261c2cb07b6c[0]
+```
+
+Similarly, if you see a Haskell error trace in the output, you have likely discovered a bug.
+
+
+### Loop-unrolling
+
+If your code contains loops, the default invocation `cargo crux-mir --lib` might be very slow. If that is the case, add `--path-sat` argument. Then each time around the loop, `crux-mir` will check with the solver whether it should keep unrolling or no. The full command is:
+
+```
+$ cargo crux-test --lib -- --path-sat
+```
+
+
+### Print counterexamples
+
+When your test fails, it is often useful to get concrete counterexample to better understand which input causes the failure. You can use `-m` argument to print the counterexample:
+
+```
+$ cargo crux-test --lib -- -m 
+...
+---- kinds_of_failure/c15b5012::overflow_quicksort[0]::crux_test[0]::midpoint_overflow[0] counterexamples ----
+[Crux] Found counterexample for verification goal
+[Crux]   src/overflow_quicksort.rs:31:27: 31:34: error: in kinds_of_failure/c15b5012::overflow_quicksort[0]::crux_test[0]::midpoint_overflow[0]
+[Crux]   attempt to compute `copy _1 + copy _2`, which would overflow
+Model:
+[{"name": "a","loc": null,"val": "-0x58cf1802","bits": "32"},{"name": "b","loc": null,"val": "0x58cf1803","bits": "32"}]
+```
+
+The model shows the name of variable, as well as its concrete hexadecimal value.
```

Comment:
Note that in general, `val` may not be printed as a hexadecimal value (that output is specific to bitvectors, I believe). Perhaps just say "its concrete value" instead.